Userland/Libraries/LibJS/Runtime/ECMAScriptFunctionObject.cpp

```diff
@@ -107,86 +108,269 @@ void ECMAScriptFunctionObject::visit_edges(Visitor& visitor)
     }
 }
 
-FunctionEnvironment* ECMAScriptFunctionObject::create_environment(FunctionObject& function_being_invoked)
+// 9.1.2.4 NewFunctionEnvironment ( F, newTarget ), https://tc39.es/ecma262/#sec-newfunctionenvironment
+FunctionEnvironment* ECMAScriptFunctionObject::new_function_environment(FunctionObject&, Object* new_target)
```

Comment:
Why is this a method but still receiving the `F` argument? That doesn't seem right - `F` should be `this`.